From 9c67a37fc7709c9e314bc56ccdf7727bee02fc92 Mon Sep 17 00:00:00 2001
From: Krzesimir Nowak <knowak@microsoft.com>
Date: Wed, 14 Dec 2022 12:52:25 +0100
Subject: [PATCH 1/2] main: Print the ignored parent line in warning

If repo name in the parent line was empty, nothing was printed.

Signed-off-by: Krzesimir Nowak <knowak@microsoft.com>
Signed-off-by: Fabian Groffen <grobian@gentoo.org>
---
 main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/main.c b/main.c
index 809a085..f6a39f9 100644
--- a/main.c
+++ b/main.c
@@ -646,6 +646,8 @@ read_portage_profile(const char *profile, env_vars vars[], set *masks)
 					repo_name = NULL;
 				}
 				if (repo_name == NULL) {
+					/* bring back the colon to see the ignored parent line */
+					*(--p) = ':';
 					warn("ignoring parent with unknown repo in profile %s: %s",
 							profile, s);
 					continue;
-- 
2.25.1

